### PR TITLE
Change schema of `mz_dataflow_channel_operators`

### PR DESCRIPTION
**Pull request by umanwizard**
The current view uses inner joins, so it does not contain channels that don't correspond to a live operator. This can occur if an operator downgrades its capabilities to empty and is therefore cleaned up by the system. Much more commonly, it can happen with channels that enter or leave a scope. By convention, these are considered to come from or to slot 0 of the scope, which has no corresponding operator.

In this PR, we add the addresses of the channels' endpoints, so that we can uniquely identify operators that have gone away and also enter/leave channels.

### CODE REVIEW

File: src/adapter/src/catalog/builtin.rs [hunk 1]
```diff
@@ -2070,36 +2070,39 @@ channel_addresses(id, worker_id, address, from_index, to_index) AS (
...
+             coa.worker_id = to_ops.worker_id
+",
 };
 
 pub const MZ_DATAFLOW_CHANNEL_OPERATORS: BuiltinView = BuiltinView {
     name: "mz_dataflow_channel_operators",
     schema: MZ_INTERNAL_SCHEMA,
     sql: "CREATE VIEW mz_internal.mz_dataflow_channel_operators AS
-SELECT id, from_operator_id, to_operator_id
+SELECT id, from_operator_id, from_operator_address, to_operator_id, to_operator_address
```
Comment (mjibson) [inline src/adapter/src/catalog/builtin.rs:2105]: Need to update mz_internal.md with the new columns.

File: doc/user/content/sql/system-catalog/mz_internal.md [hunk 2]
```diff
@@ -404,11 +404,13 @@ Field           | Type       | Meaning
...
+| Field                   | Type            | Meaning                                                                                                             |
+|-------------------------|-----------------|---------------------------------------------------------------------------------------------------------------------|
+| `id`                    | [`bigint`]      | The ID of the channel. Corresponds to [`mz_dataflow_channels.id`](#mz_dataflow_channels).                           |
+| `from_operator_id`      | [`bigint`]      | The ID of the source of the channel. Corresponds to [`mz_dataflow_operators.id`](#mz_dataflow_operators).           |
+| `from_operator_address` | [`bigint list`] | The address of the source of the channel. Corresponds to [`mz_dataflow_addresses.address`](#mz_dataflow_addresses). |
+| `to_operator_id`        | [`bigint`]      | The ID of the target of the channel. Corresponds to [`mz_dataflow_operators.id`](#mz_dataflow_operators).           |
+| `to_operator_address`   | [`bigint list`] | The address of the target of the channel. Corresponds to [`mz_dataflow_addresses.address`](#mz_dataflow_addresses). |
```
Comment (teskje) [inline doc/user/content/sql/system-catalog/mz_internal.md:413]: Nit: The formatting style of this table now is different from all other tables in this doc. No idea if the docs team cares, but it bugs me!

**Review by teskje — APPROVED** [review]
LGTM, except for one docs formatting nit.

File: doc/user/content/sql/system-catalog/mz_internal.md (hunk 2 above)
Comment (umanwizard) [reply doc/user/content/sql/system-catalog/mz_internal.md:413]: Actually, there are some tables formatted this way already (`mz_dataflow_arrangement_sizes`), and some the other way, so this is not adding any inconsistency.

My editor uses this format automatically; unfortunately it doesn't seem to be configurable.

Comment (umanwizard) [reply doc/user/content/sql/system-catalog/mz_internal.md:413]: I have reformatted all tables in this style, for consistency.

**Pull request merged**